spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -883,6 +916,23 @@ private void checkIdle() {
 			}
 		}
 
+		private void idleBetweenPollIfNecessary() {
+			long idleBetweenPolls = this.containerProperties.getIdleBetweenPolls();
+			if (idleBetweenPolls > 0) {
+				idleBetweenPolls = Math.min(idleBetweenPolls,
+						this.maxPollInterval - (System.currentTimeMillis() - this.lastPoll));
```

Comment:
`this.lastPoll` is when the last poll exited.

I think there might be a problem with a large `pollTimeout` because I think `max.poll.interval` is the time between polls, not the time between when the last poll exited and the next poll.

Since `lastPoll` is only used (otherwise) for idle consumer detection, I think it is safe to move it to before the poll.

Reply:
Resolved in the latest commit. See its description.

Thank you for feedback!